src/hooks/swap/useSwapInfo.tsx

```diff
@@ -34,28 +35,27 @@ interface SwapInfo {
 // from the current swap inputs, compute the best trade and return it.
 function useComputeSwapInfo(routerUrl?: string): SwapInfo {
   const isWrap = useIsWrap()
-  const { independentField, amount, [Field.INPUT]: currencyIn, [Field.OUTPUT]: currencyOut } = useAtomValue(swapAtom)
-  const isExactIn = independentField === Field.INPUT
+  const { type, amount, [Field.INPUT]: currencyIn, [Field.OUTPUT]: currencyOut } = useAtomValue(swapAtom)
 
   const parsedAmount = useMemo(
-    () => tryParseCurrencyAmount(amount, (isExactIn ? currencyIn : currencyOut) ?? undefined),
-    [amount, isExactIn, currencyIn, currencyOut]
+    () => tryParseCurrencyAmount(amount, (isExactInput(type) ? currencyIn : currencyOut) ?? undefined),
+    [amount, type, currencyIn, currencyOut]
   )
   const hasAmounts = currencyIn && currencyOut && parsedAmount && !isWrap
   const trade = useRouterTrade(
-    isExactIn ? TradeType.EXACT_INPUT : TradeType.EXACT_OUTPUT,
+    type,
     routerUrl,
     hasAmounts ? parsedAmount : undefined,
-    hasAmounts ? (isExactIn ? currencyOut : currencyIn) : undefined
+    hasAmounts ? (isExactInput(type) ? currencyOut : currencyIn) : undefined
```

Comment:
Nit: can `isExactInput(type)` be moved to a constant here and re-used?

Reply:
(as we use it throughout the file?)

Reply:
I'd like to keep it with the helper function because it's just a boolean check, and doing so keeps it consistent (so - hopefully - easier to read).